feedback-module/src/components/Module.js

```diff
@@ -146,28 +96,32 @@ function Module({ pagetitle, endpoint, dir }) {
       setCheckboxError(true);
       firstCheckRef.current && firstCheckRef.current.focus();
       // Make sure all required fields are completed
-    } else if (
-      screen.textInputs &&
-      !inputsValidated(inputQuestions, setInputQuestions)
-    ) {
-      const firstErrorIndex = inputQuestions.findIndex(
-        (question) => question.error
-      );
-      if (firstErrorIndex >= 0 && inputRefs[firstErrorIndex].current) {
-        inputRefs[firstErrorIndex].current.focus();
-      }
+    } else if (screen.textInputs && !inputsValidated()) {
+      focusFirstError();
     } else {
-      screen.formID && handleSubmit(),
+      updateOtherField();
+      sendFormData(screen.formID),
         setScreen(SCREENS[nextScreen]),
         setCheckboxError(false);
     }
-    headerRef.current.scrollIntoView(true);
   };
 
-  const onCheck = (index) => {
-    let checked = checkedFields;
-    checked[index].checked = !checked[index].checked;
-    setCheckedFields(checked);
+  const handleSubmit = (e) => {
+    e.preventDefault();
+  };
+
+  const changeScreen = (text, nextScreen, feedbackID) => {
+    // If button contains a feedbackID, update the feedbackType of the feedback object
+    if (screen.formID) {
+      submitForm(nextScreen);
+    } else {
+      feedbackID &&
+        updateFeedbackForAPI(setFeedbackType, [en(text), feedbackID]);
```

Comment:
Why are we still updating the feedback for the API here?

Reply:
If this is where we are setting this initial option that they checked then we should probably put it in an else if because we only will need to do this once and not every time there isn't a form id

Reply:
Yes— there's another check here to see if the button does have a feedbackID (which is only true for the buttons on the initial page), so this only gets set once.